src/core/http/request_handler.rs

```diff
@@ -179,25 +169,17 @@ async fn handle_request_with_cors<T: DeserializeOwned + GraphQLRequestLike>(
     app_ctx: Arc<AppContext>,
     request_counter: &mut RequestCounter,
 ) -> Result<Response<Body>> {
-    // Safe to call `.unwrap()` because this method will only be called when
-    // `cors` is `Some`
     let cors = app_ctx.blueprint.server.cors.as_ref().unwrap();
     let (parts, body) = req.into_parts();
     let origin = parts.headers.get(&header::ORIGIN);
 
     let mut headers = HeaderMap::new();
-
-    // These headers are applied to both preflight and subsequent regular CORS
-    // requests: https://fetch.spec.whatwg.org/#http-responses
-
```

Comment:
Let's add the comments back

Reply:
> Let's add the comments back

Okay will add and commit it

Reply:
Also it would be better if we can use some sort of middleware API and move the logic into independent middleware. I think that would be a much better design. Hopefully no degradation in performance.

Reply:
Can we solve it in another issue? It's just my opinion 😅. I think we need good implementation plan for it and I will work on it by just bit research of how we can implement without any performance degradation :)